packages/application-extension/src/index.ts

```diff
@@ -172,26 +169,24 @@ const router: JupyterLabPlugin<IRouter> = {
       PageConfig.getBaseUrl(),
       PageConfig.getOption('pageUrl')
     );
-    const tree = PageConfig.getTreeUrl();
     const router = new Router({ base, commands });
+    const pattern = /^\/tree\/(.*)/;
 
     commands.addCommand(CommandIDs.tree, {
       execute: (args: IRouter.ICommandArgs) => {
-        const path = (args.path as string).replace('/tree', '');
+        return app.restored.then(() => {
+          const path = decodeURIComponent((args.path.match(pattern)[1]));
```

Comment:
Does this now handle filenames and paths with spaces?

Reply:
Encode/decode URI component fixes spaces.